eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -1040,88 +1040,91 @@ object Helpers {
-        // now we know what commit number this tx is referring to, we can derive the commitment point from the shachain
-        remotePerCommitmentSecrets.getHash(0xFFFFFFFFFFFFL - txNumber)
-          .map(d => PrivateKey(d))
-          .map(remotePerCommitmentSecret => {
-            val remotePerCommitmentPoint = remotePerCommitmentSecret.publicKey
-            val remoteDelayedPaymentPubkey = Generators.derivePubKey(remoteParams.delayedPaymentBasepoint, remotePerCommitmentPoint)
-            val remoteRevocationPubkey = Generators.revocationPubKey(keyManager.revocationPoint(channelKeyPath).publicKey, remotePerCommitmentPoint)
-            val remoteHtlcPubkey = Generators.derivePubKey(remoteParams.htlcBasepoint, remotePerCommitmentPoint)
-            val localPaymentPubkey = Generators.derivePubKey(keyManager.paymentPoint(channelKeyPath).publicKey, remotePerCommitmentPoint)
-            val localHtlcPubkey = Generators.derivePubKey(keyManager.htlcPoint(channelKeyPath).publicKey, remotePerCommitmentPoint)
-
-            val feeratePerKwMain = feeEstimator.getFeeratePerKw(feeTargets.claimMainBlockTarget)
-            // we need to use a high fee here for punishment txs because after a delay they can be spent by the counterparty
-            val feeratePerKwPenalty = feeEstimator.getFeeratePerKw(target = 2)
-
-            // first we will claim our main output right away
-            val mainTx = channelFeatures match {
-              case ct if ct.paysDirectlyToWallet =>
-                log.info(s"channel uses option_static_remotekey to pay directly to our wallet, there is nothing to do")
-                None
-              case ct => ct.commitmentFormat match {
-                case DefaultCommitmentFormat => withTxGenerationLog("claim-p2wpkh-output") {
-                  Transactions.makeClaimP2WPKHOutputTx(commitTx, localParams.dustLimit, localPaymentPubkey, localParams.defaultFinalScriptPubKey, feeratePerKwMain).map(claimMain => {
-                    val sig = keyManager.sign(claimMain, keyManager.paymentPoint(channelKeyPath), remotePerCommitmentPoint, TxOwner.Local, commitmentFormat)
-                    Transactions.addSigs(claimMain, localPaymentPubkey, sig)
-                  })
-                }
-                case _: AnchorOutputsCommitmentFormat => withTxGenerationLog("remote-main-delayed") {
-                  Transactions.makeClaimRemoteDelayedOutputTx(commitTx, localParams.dustLimit, localPaymentPoint, localParams.defaultFinalScriptPubKey, feeratePerKwMain).map(claimMain => {
-                    val sig = keyManager.sign(claimMain, keyManager.paymentPoint(channelKeyPath), TxOwner.Local, commitmentFormat)
-                    Transactions.addSigs(claimMain, sig)
-                  })
+        if (txNumber > 0xffffffffffffL) {
+          // txNumber must be lesser than 48 bits long
+          None
+        } else {
+          log.warning(s"a revoked commit has been published with txnumber=$txNumber")
```

Comment:
nit: this log line should go inside the last `map`, we only know that it's a revoked commit once we've been able to extract the per-commitment secret

Reply:
Done in 2cb3262db.